stores/moderationSubmissionsListStore.ts

```diff
@@ -0,0 +1,249 @@
+            facilityData.value = facilityResults
+        }
+
+        async function getHealthcareProfessionals() {
+            const healthcareProfessionalResults = await queryHealthcareProfessionals()
+            healthcareProfessionalsData.value = healthcareProfessionalResults
+        }
+
+        function setSelectedModerationListViewChosen(selectedOption: SelectedModerationListView) {
+            selectedModerationListViewChosen.value = selectedOption
+        }
+
+        function filterSelectedSubmission(submissionId: string | undefined) {
+            selectedSubmissionData.value = submissionsData.value.find(submission => submission.id === submissionId)
+        }
+
+        function filterSubmissionsBySelectedTab(tabValue: SelectedSubmissionListViewTab) {
+            switch (tabValue) {
+                case SelectedSubmissionListViewTab.ForReview:
+                    filteredSubmissionDataForListComponent.value = submissionsData.value
+                        .filter((submission: Submission) => {
+                            const isNewSubmission = !submission.isRejected && !submission.isApproved && !submission.isUnderReview
+                            return submission.isUnderReview || isNewSubmission
+                        })
+                    break
+                case SelectedSubmissionListViewTab.Approved:
+                    filteredSubmissionDataForListComponent.value = submissionsData.value
+                        .filter((submission: Submission) => submission.isApproved)
+                    break
+                case SelectedSubmissionListViewTab.Rejected:
+                    filteredSubmissionDataForListComponent.value = submissionsData.value
+                        .filter((submission: Submission) => submission.isRejected)
+                    break
+            }
+        }
+        return { getSubmissions,
+            submissionsData,
+            filterSubmissionsBySelectedTab,
+            filteredSubmissionDataForListComponent,
+            selectedSubmissionId,
+            filterSelectedSubmission,
+            selectedSubmissionData,
+            getFacilities,
+            facilityData,
+            getHealthcareProfessionals,
+            healthcareProfessionalsData,
+            selectedModerationListViewChosen,
+            setSelectedModerationListViewChosen,
+            selectedFacilityId,
+            selectedHealthcareProfessionalId }
+    }
+)
+
+async function querySubmissions() {
+    try {
+        const submissionsFilters = {
+            filters: {
+                id: undefined
+            }
+        }
+
+        const result = await gqlClient.request<{ submissions: Submission[] }>(
+            getSubmissionsGqlQuery,
+            submissionsFilters
+        )
+        return result?.submissions ?? []
+    } catch (error) {
+        console.log(`Error querying the submissions: ${JSON.stringify(error)}`)
+        return []
+    }
+}
+
+async function queryFacilities() {
```

Comment:
I think it would be good to move the queryFacilities and queryProfessionals to their own respective state stores and then call them from here.